embedded/drivers/src/analog.cpp

```diff
@@ -0,0 +1,58 @@
+#include "analog.h"
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#include <fstream>
+#include <iostream>
+
+#define NUM_ANALOG_PINS 7
+#define PIN_FILE_PATH "/sys/bus/iio/devices/iio:device0/in_voltage%d_raw"
+#define BBB_ADC_RANGE 1.8
+#define BBB_ADC_RESOLUTION 4096.0
+
+AnalogPin::AnalogPin(int pin, float scale_factor) {
+  this->scale_factor = scale_factor;
+  this->pin = pin;
+}
+
+AnalogPin::~AnalogPin() { close(this->fd); }
+
+bool AnalogPin::begin() {
+  char filePath[49];
+
+  if (this->pin < 0 || this->pin >= NUM_ANALOG_PINS) {
+    std::cout << "Pin number not in range\n";
+    return false;
+  }
+
+  if (this->scale_factor <= 0) {
+    std::cout << "Scale Factor must be positive\n";
+    return false;
+  }
+
+  sprintf(filePath, PIN_FILE_PATH, this->pin);
```

Comment:
I understand this but it just seems a bit confusing to me. Maybe something like

```c++
#define PIN_NUMBER_TO_FILE_PATH(x, y) sprintf(x, PIN_FILE_PATH, y)
```

Not sure, lmk your thoughts.

Reply:
That makes sense. Added the macro w/ meaningful parameter names